tsconfig.json

```diff
@@ -19,6 +19,7 @@
     "resolveJsonModule": true,
     "sourceMap": true,
     "strict": true,
+    "strictPropertyInitialization": false,
```

Comment:
Please do not do this.  We should not be working around bad code by making TypeScript less strict.

Reply:
I agree actually but what happened is that last time, it never showed the errors `Property 'referrerPolicy' has no initializer and is not definitely assigned in the constructor.` everywhere until now. So this was the only workaround. I'll revert it. Sorry!

Reply:
Also please review the remaining stuff, how good (or bad) is it? Also I had an issue with `headers` in `fetch.ts`'s `fetch` method. If you could please look into it.